src/instructlab/data/generate.py

```diff
@@ -228,8 +229,33 @@ def generate(
         batch_size = 8
 
     backend_instance = None
+    # check the user's host ports to see if anything is actively running there,
+    # priority is given by the ordering of the hostports
+    user_host_ports = [
+        ctx.obj.config.generate.teacher.host_port,
+        ctx.obj.config.serve.host_port,
+    ]
+    active_url: str | None = None
+    for hp in user_host_ports:
+        api_base = get_api_base(hp)
+        if is_openai_server_and_serving_model(
+            api_base,
+            api_key=api_key,
+            http_params={
+                "tls_insecure": tls_insecure,
+                "tls_client_cert": tls_client_cert,
+                "tls_client_passwd": tls_client_passwd,
+                "tls_client_key": tls_client_key,
+            },
+        ):
+            active_url = api_base
+
     if endpoint_url:
         api_base = endpoint_url
+    elif active_url:
```

Comment:
Is it possible that we'd be serving the wrong model? i.e. we want to be serving Mixtral but we're serving Mistral on that port. We might want to identify _which_ model is being served and ask the user if they want to use that, or if that server should be shut down and a new one booted.

Reply:
This is true and could actually become a common mistake. I'll introduce some logic here to verify that this is the indended model to use. If not, we'll just continue on with the rest of the function. 

The only concern I have here is that the model we prompt will be in the format of `/very/long/path/to/.local/share/instructlab/models/mixtral` - should this be presented as-is, or should we trim out the base path if we detect that the model is in this format?

Reply:
I mean, could totally do:
```
click.secho(f"Name of model currently served: {active_url.split("/")[-1]}")
click.secho(f"This model's path is: {active_url}")
click.confirmation("If you would like to keep using this server, please confirm. Otherwise, please shut down this server and try again.", abort=True)
```

Reply:
Discussed offline, but the other consideration that we need to make here is when `--non-interactive` is passed, we will skip finding the server automatically and provide a warning in red about the fact we skipped this check. We will include a recommendation here that the user needs to be explicit about the teacher model.